src/object/srv_obj.c

```diff
@@ -3356,22 +3356,24 @@ obj_punch_agg_cb(struct dtx_leader_handle *dlh, void *agg_arg)
 {
 	uint64_t	*flag = agg_arg;
 	uint32_t	sub_cnt = dlh->dlh_normal_sub_cnt + dlh->dlh_delay_sub_cnt;
-	int		succeeds = 0;
 	int		allow_failure = 0;
 	int		allow_failure_cnt = 0;
-	int		result = 0;
+	int		result = dlh->dlh_result;
```

Comment:
Hmm, this function will be called when all sub and local has finished, i.e. all future has been set by (obj_tgt_punch), since it will only be called once by one helper or the rpc handler ULT after all sub ULTs have been finished, so it seems good to me. @Nasf-Fan